src/analysis/retail/priest/holy/modules/talents/MiddleRow/EverlastingLight.tsx

```diff
@@ -0,0 +1,80 @@
+import { formatNumber, formatPercentage } from 'common/format';
+import TALENTS from 'common/TALENTS/priest';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent, HealEvent } from 'parser/core/Events';
+import SPELLS from 'common/SPELLS';
+import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+
+const MAX_EVERLASTING_LIGHT_BUFF = 0.15;
+
+class EverlastingLight extends Analyzer {
+  currentHealingBonus: number = 0;
+  rawAdditionalHealing: number = 0;
+  effectiveAdditionalHealing: number = 0;
+  overhealing: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS.EVERLASTING_LIGHT_TALENT.id);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.GREATER_HEAL), this.onCast);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.GREATER_HEAL), this.onHeal);
+  }
+
+  get percentOverhealing() {
+    if (this.rawAdditionalHealing === 0) {
+      return 0;
+    }
+    return this.overhealing / this.rawAdditionalHealing;
+  }
+
+  onCast(event: CastEvent) {
+    event.classResources &&
+      event.classResources.forEach((resource) => {
+        if (resource.type !== RESOURCE_TYPES.MANA.id) {
+          return;
+        }
+        const currentMana = resource.amount;
+        this.currentHealingBonus =
+          MAX_EVERLASTING_LIGHT_BUFF * Math.max(0, 1 - currentMana / resource.max);
+      });
+  }
```

Comment:
should probably use `ManaTracker` for this, which handles some edge cases. not sure if those edge cases are relevant for hpriest (they're mostly around multiple rapid updates), but you could drop this cast tracking and instead use `this.manaTracker.current / this.manaTracker.maxResource` in the `onHeal`.

Reply:
(this is optional, won't block the PR, but might be easier to maintain longterm)